autode/opt/coordinates/dic.py

```diff
@@ -0,0 +1,239 @@
+"""
+Delocalised internal coordinate implementation from:
+1. https://aip.scitation.org/doi/pdf/10.1063/1.478397
+and references cited therein. Also used is
+2. https://aip.scitation.org/doi/pdf/10.1063/1.1515483
+
+The notation follows the paper and is briefly
+summarised below:
+
+| x : Cartesian coordinates
+| B : Wilson B matrix
+| G : 'Spectroscopic G matrix'
+| q : Redundant internal coordinates
+| s : Non-redundant internal coordinates
+| U : Transformation matrix q -> s
+|
+"""
+import numpy as np
+from time import time
+from typing import Optional
+from autode.geom import rotate_columns
+from autode.log import logger
+from autode.opt.coordinates.primitives import ConstrainedDistance
+from autode.opt.coordinates.internals import (PIC,
+                                              InverseDistances,
+                                              InternalCoordinates)
+
+
+class DIC(InternalCoordinates):  # lgtm [py/missing-equals]
+    """Delocalised internal coordinates"""
+
+    def __repr__(self):
+        return f'DIC(n={len(self)})'
+
+    @staticmethod
+    def _calc_U(primitives: PIC) -> np.ndarray:
+        r"""
+        Transform matrix containing the non-redundant eigenvectors of the G
+        matrix.
+
+        .. math::
+
+            G (U R) = (U R) \begin{pmatrix}
+            \Lambda & 0 \\
+            0 & 0
+            \end{pmatrix}
+
+        where
+
+        .. math::
+
+            G = B B^{T}
+
+
+        -----------------------------------------------------------------------
+        Arguments:
+            primitives (autode.opt.internals.PIC):
+
+        Returns:
+            (np.ndarray): U
+        """
+
+        w, v = np.linalg.eigh(primitives.G)
+
+        # Form a transform matrix from the primitive internals to a set of
+        # 3N - 6 non-redundant internals, s
+        v = v[:, np.where(np.abs(w) > 1E-10)[0]]
```

Comment:
Could be obvious to a domain expert, but the comment does'nt really help me unpack that line.

Reply:
I'm not sure it would be obvious 😕. Thanks - will try and improve

Reply:
Always hard to  think about these things when you are deep in coding your methods :)